resolver-dns/src/main/java/io/netty/resolver/dns/DnsResolveContext.java

```diff
@@ -440,20 +505,22 @@ private boolean handleRedirect(
             AuthoritativeNameServerList serverNames = extractAuthoritativeNameServers(question.name(), res);
 
             if (serverNames != null) {
-                List<InetSocketAddress> nameServers = new ArrayList<InetSocketAddress>(serverNames.size());
+                Set<InetSocketAddress> resolvedNameServers = parent.newNameServerSet();
```

Comment:
using a set may re-order the results provided by the server, which may be in preference order from the server's perspective. Should we just depend upon the `uncachedRedirectDnsServerList` below to apply preference/sorting order so the user can see the original order and respect it if desired?

Reply:
@Scottmitch good point.